snapcraft/extractors/appstream.py

```diff
@@ -194,6 +194,10 @@ def _get_icon_from_desktop_file(
         entry = DesktopEntry()
         entry.parse(os.path.join(workdir, path))
         icon = entry.getIcon()
+
+        if icon.startswith("${{SNAP}}"):  # matches '${SNAP}'.
+            icon = icon[7:]  # strip the '${SNAP}' prefix.
```

Comment:
you also need to strip the leading slash for "joins" to work

Reply:
Let's see how the testsuite copes with that then. Changeset 4e56055 adds a trailing slash to the `${SNAP}`-stripping code.